src/test/java/com/cloudant/tests/util/Utils.java

```diff
@@ -66,13 +73,27 @@ public static String getHostName(String account) {
     public static void removeReplicatorTestDoc(CloudantClient account, String replicatorDocId)
             throws Exception {
 
-        //Grab replicator doc revision using HTTP GET command
+        //Grab replicator doc revision using HTTP HEAD command
         String replicatorDb = "_replicator";
         URI uri = URI.create(account.getBaseUri() + replicatorDb + "/"
                 + replicatorDocId);
         HttpConnection head = Http.HEAD(uri);
-        String revision = head.execute().getConnection().getHeaderField("ETag");
 
+        //add a response interceptor to allow us to retrieve the ETag revision header
+        final AtomicReference<String> revisionRef = new AtomicReference<String>();
```

Comment:
I'm not familiar with the `AtomicReference` class but is it needed? I thought the method in the anonymous `HttpConnectionResponseInterceptor` class would capture the `revisionRef` variable and that would be sufficient.